buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -82,7 +82,7 @@ object Java9Modularity {
                         )
 
                         // add the resulting module descriptor to this target's artifact
-                        artifactTask.from(compileModuleTask) {
+                        artifactTask.from(compileModuleTask.map { it.destinationDirectory }) {
```

Comment:
Does calling `jar` tasks (artifactTask) result in calling `compileModuleTask`?
This should work, but in some versions of Gradle it sometimes worked incorrectly.

Reply:
I guess so, it is a default java compilation task.

Reply:
Checked locally, calling `jar` leads to running `compileKotlinJvmModule`